src/api/layer/authenticated_http.ts

```diff
@@ -1,12 +1,21 @@
 import { version as packageVersion } from '../../../package.json'
-import { APIError } from '../../models/APIError'
+import { APIError, APIErrorMessage } from '../../models/APIError'
 import { reportError } from '../../models/ErrorHandler'
 
 const CUSTOM_PREFIX = 'Layer-'
 const CUSTOM_HEADERS = {
-  [`${CUSTOM_PREFIX}React-Version`]: packageVersion
+  [`${CUSTOM_PREFIX}React-Version`]: packageVersion,
 } as const
 
+type APIResponseError = {
+  description?: string
+  type?: string
+}
+
+type FailedAPIResponse = {
+  errors?: APIResponseError[]
+}
```

Comment:
Main change - improve types of API responses - because it comes from outside, all props are optional